lib/relations/Relation.js

```diff
@@ -151,14 +155,14 @@ class Relation {
       ownerTable = builder.tableRefFor(this.ownerModelClass.getTableName())
     } = {}
   ) {
-    let relatedSelect = relatedJoinSelectQuery.modify(this.modify).as(relatedTableAlias);
+    let relatedJoinSelect = this.applyModify(relatedJoinSelectQuery).as(relatedTableAlias);
 
-    if (relatedSelect.isSelectAll()) {
+    if (relatedJoinSelect.isSelectAll()) {
       // No need to join a subquery if the query is `select * from "RelatedTable"`.
-      relatedSelect = aliasedTableName(relatedTable, relatedTableAlias);
+      relatedJoinSelect = aliasedTableName(relatedTable, relatedTableAlias);
     }
 
-    return builder[joinOperation](relatedSelect, join => {
+    return builder[joinOperation](relatedJoinSelect, join => {
```

Comment:
@koskimas I've renamed these for consistency with the same code in `ManyToManyRelation`